corehq/ex-submodules/pillowtop/pillow/interface.py

```diff
@@ -176,6 +192,7 @@ def process_offset_chunk(chunk, context):
                         self._update_checkpoint(change, context)
                 else:
                     self._update_checkpoint(None, None)
+            context.flush_checkpoint_on_next_opportunity()
             process_offset_chunk(changes_chunk, context)
         except PillowtopCheckpointReset:
             process_offset_chunk(changes_chunk, context)
```

Comment:
I think you can remove the next line completely since this is now called in a loop
```diff
         except PillowtopCheckpointReset:
             process_offset_chunk(changes_chunk, context)
-            self.process_changes(since=self.get_last_checkpoint_sequence(), forever=forever)
```

Reply:
Ah thanks, yeah that makes sense